arch/openrisc/lib/memset.S

```diff
@@ -0,0 +1,95 @@
+/*
+ * OpenRISC memset.S
+ *
+ * Hand-optimized assembler version of memset for OpenRISC.
+ * Algorithm inspired by several other arch-specific memset routines
+ * in the kernel tree
+ *
+ * Copyright (C) 2015 Olof Kindgren <olof.kindgren@gmail.com>
+ *
+ *      This program is free software; you can redistribute it and/or
+ *      modify it under the terms of the GNU General Public License
+ *      as published by the Free Software Foundation; either version
+ *      2 of the License, or (at your option) any later version.
+ */
+
+	.global memset
+	.type	memset, @function
+memset:
+	#Args r3 = *s, r4 = c, r5 = n
+	#r13, r15, r17, r19 used as temp regs
+
+	#Exit if n == 0
+	l.sfeqi		r5, 0
+	l.bf		4f
+
+	#Truncate c to char
+	l.andi  	r13, r4, 0xff
+
+	#Skip word extension if c is 0
+	l.sfeqi		r13, 0
+	l.bf		1f
+	#Check for at least two whole words (8 bytes) in delay slot
```

Comment:
Aha, clever. Hadn't seen that before but I like it. Just to be sure, should we always use this for instructions after branches, or just for cases where we rely on the delay slot?